### PR TITLE
Add min supported version check for isValidReleaseVersion

### PR DESCRIPTION
**Pull request by enxebre**
**What this PR does / why we need it**:
Add min supported version check for isValidReleaseVersion

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Review by csrwng — COMMENTED** [review]
Thank you @enxebre !
Just one comment.

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go [hunk 1]
```diff
@@ -3559,19 +3559,25 @@ func isProgressing(ctx context.Context, hc *hyperv1.HostedCluster) (bool, error)
...
 	if networkType == hyperv1.OpenShiftSDN && currentVersion != nil && currentVersion.Minor < version.Minor {
 		return fmt.Errorf("y-stream upgrade is not for OpenShiftSDN")
 	}
 
 	if (version.Major == latestVersionSupported.Major && version.Minor > latestVersionSupported.Minor) || version.Major > latestVersionSupported.Major {
-		return fmt.Errorf("the latest HostedCluster supported by this Operator is: %q. Attempting to use: %q", supportedversion.LatestSupportedVersion, version.Major)
+		return fmt.Errorf("the latest HostedCluster version supported by this Operator is: %q. Attempting to use: %q", supportedversion.LatestSupportedVersion, version.Major)
+	}
+
+	if version.LE(*minSupportedVersion) {
```
[the remark this answers is not in the document]
Comment (sjenning) [reply hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:3579]: Ha @csrwng beat me

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**